network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,438 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Etienne Carriere <etienne.carriere@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: bigip_virtual_server
+short_description: "Manages F5 BIG-IP LTM virtual servers"
+description:
+    - "Manages F5 BIG-IP LTM virtual servers via iControl SOAP API"
+version_added: "2.0"
+author: Etienne Carriere
+notes:
+    - "Requires BIG-IP software version >= 11"
+    - "F5 developed module 'bigsuds' required (see http://devcentral.f5.com)"
+    - "Best run as a local_action in your playbook"
+requirements:
+    - bigsuds
+options:
+    server:
+        description:
+            - BIG-IP host
+        required: true
+    user:
+        description:
+            - BIG-IP username
+        required: true
+    password:
+        description:
+            - BIG-IP password
+        required: true
+    validate_certs:
+        description:
+            - If C(no), SSL certificates will not be validated. This should only be used
+              on personally controlled sites using self-signed certificates.
+        required: false
+        default: 'yes'
+        choices: ['yes', 'no']
+    state:
+        description:
+            - Pool member state
+        required: false
+        default: present
+        choices: ['present', 'absent', 'enabled', 'disabled']
+        aliases: []
+    partition:
+        description:
+            - Partition
+        required: false
+        default: 'Common'
+    name:
+        description:
+            - "Virtual server name."
+        required: true
+        aliases: ['vs']
+    destination:
+        description:
+            - "Destination IP of the virtual server (only host is currently supported) . Required when state=present and vs does not exist. Error when state=absent."
```

Comment:
Corrected the documentation . It is right it is not an issue to have destination with state=absent . It is just ignored.